tst/providers/MeetingProvider/MeetingManager.test.tsx

```diff
@@ -0,0 +1,137 @@
+    });
+
+    it('should call addObserver on AudioVideoFacade', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      expect(mockMeetingManager?.audioVideo?.addObserver).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call setupAudioVideoObservers', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      expect(mockMeetingManager.setupAudioVideoObservers).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call setupDeviceLabelTrigger', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      expect(mockMeetingManager.setupDeviceLabelTrigger).toHaveBeenCalledTimes(1);
+    });
+    
+    it('should call listAndSelectDevices', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      expect(mockMeetingManager.listAndSelectDevices).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call publishAudioVideo', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      expect(mockMeetingManager.publishAudioVideo).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call setupActiveSpeakerDetection', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      expect(mockMeetingManager.setupActiveSpeakerDetection).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call publishMeetingStatus', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      // @ts-ignore
+      expect(mockMeetingManager.publishMeetingStatus).toHaveBeenCalledTimes(1);
+    });
+
+    it('should use the MeetingSessionConfiguration provided via params', async () => {
+      mockMeetingManager.meetingSessionConfig = null;
+      
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      expect(mockMeetingManager.meetingSessionConfig).not.toBeNull();
+    });
+
+    it('should use simulcastEnabled from MeetingManagerConfig to update MeetingSessionConfiguration', async () => {
+      mockMeetingManagerConfig = {
+        logLevel: LogLevel.WARN,
+        simulcastEnabled: true
+      };
+      mockMeetingManager = new MeetingManager(mockMeetingManagerConfig);
+      mockMeetingManager.setupAudioVideoObservers = jest.fn().mockReturnValue({});
+      mockMeetingManager.setupDeviceLabelTrigger = jest.fn().mockReturnValue({});
+      mockMeetingManager.listAndSelectDevices = jest.fn().mockReturnValue({});
+      mockMeetingManager.publishAudioVideo = jest.fn().mockReturnValue({});
+      mockMeetingManager.setupActiveSpeakerDetection = jest.fn().mockReturnValue({});
+      // @ts-ignore
+      mockMeetingManager.publishMeetingStatus = jest.fn().mockReturnValue({});
+
+      await mockMeetingManager.join(mockMeetingJoinData);
+  
+      expect(mockMeetingManager?.meetingSessionConfig?.enableUnifiedPlanForChromiumBasedBrowsers).toBe(true);
+      expect(mockMeetingManager?.meetingSessionConfig?.enableSimulcastForUnifiedPlanChromiumBasedBrowsers).toBe(true);
+    });
+  });
+});
```

Comment:
nit: new line end of file.
Just curious did you run `npm run lint`? If so we might need to update eslint config to add this rule.

Reply:
Forgot to run lint. Done now.

Reply:
Could we just add it to build:release step?

Reply:
I added it but it led to a bunch of changes in the PR. Not sure if it is a great idea to do that right now. We will need to first resolve all of the outstanding linting issues and can start from there. @devalevenkatesh's comment:
>Can you please run npm run lint once with the PR? We do not run it as part of npm run build:release since it results in auto-fixing and the check fails for un-committed changes.